pam_ssh_agent_auth: 0.10.3 -> 0.10.4

wmertens opened the pull request:
###### Motivation for this change

Fixes #83870 (hopefully)

###### Things done

found maintained fork, updated multi key patch (manually), didn't actually test yet

edolstra commented on pkgs/os-specific/linux/pam_ssh_agent_auth/default.nix:
```diff
@@ -1,31 +1,22 @@
 
 stdenv.mkDerivation rec {
-  name = "pam_ssh_agent_auth-0.10.3";
+  name = "pam_ssh_agent_auth-git-master";
```
Strings like `git-master` should be avoided since they're not a version (they don't start with a digit). E.g. `nix-env` won't consider this the same package anymore.

wmertens replied:
The problem is that they didn't update the version, so their version number is lower than the original but they have more fixes :-( should I just use their version?

vcunat replied:
Probably.  I don't think this will be commonly installed by `nix-env`, and on other places you don't care about version "downgrade" AFAIK.

edolstra replied:
You can use a version like `20200612` or whatever the date of git master is.

worldofpeace replied:
Just a little correction, that the manual says to do like `2020-06-12` with dashes

wmertens commented:
I somehow missed that the original author is semi-maintaining his github repo at https://github.com/jbeverly/pam_ssh_agent_auth

I updated some issues there and added a PR for Eelco's multi-keyfile patch. I'll wait for a bit, hopefully he'll answer soon.

@edolstra it turns out that the segfault is due to using EDCSA keys

wmertens committed:
pam_ssh_agent_auth: 0.10.3 -> 0.10.4

wmertens commented:
Ok, the official upstream is updated now, so I was able to build it using the regular release.

wmertens commented:
Well, since this works and fixes an issue, I'll just merge.
